aquarius/app/assets.py

```diff
@@ -705,6 +705,126 @@ def update_ratings(did):
+              description: Signature using updated field to verify that the consumer has rights to update onwership
+              type: string
+              example: "0x42e940108a430b91796341e29001319b2b2c4743156cdbe0e17afdae82b4cf9a7e1b4e6\
+41cd57d8f087ab6432cc9e53989f3ce121b6897fa3f594e9753c4ea331b"
+            "updated":
+              description: Last update field of the DDO
+              type: string
+              example: "2020-01-01T00:00:00Z"
+            "serviceIndex":
+              description: Index of the services array for the compute service to be updated
+              type: int
+              example: 2
+            "allowRawAlgorithm":
+              description: allow Raw Algorithm on this service
+              type: boolean
+              example: true
+            "allowNetworkAccess":
+              description: allow network access for the algorithm pod
+              type: boolean
+              example: true
+            "trustedAlgorithms":
+              description: A list of trusted Algorithms DIDs
+              type: array
+              example: ["did:op:123","did:op:1234"]
+    responses:
+      200:
+        description: Asset updated
+      400:
+        description: One of the required attributes is missing.
+      401:
+        description: Not authorized.
+      404:
+        description: Invalid asset data.
+      500:
+        description: Error
+    """
+    data = request.json
+    required_attributes = [
+        'signature',
+        'updated',
+        'serviceIndex',
+        'allowRawAlgorithm',
+        'allowNetworkAccess',
+        'trustedAlgorithms'
+    ]
+    msg, status = check_required_attributes(
+        required_attributes, data, 'update_computeprivacy')
+    if msg:
+        return msg, status
+    if not isinstance(data['allowRawAlgorithm'], bool):
+        logger.error('allowRawAlgorithm is not boolean')
+        return f'allowRawAlgorithm is not boolean', 400
+    if not isinstance(data['allowNetworkAccess'], bool):
+        logger.error('allowNetworkAccess is not boolean')
+        return f'allowNetworkAccess is not boolean', 400
+    if not isinstance(data['serviceIndex'], int):
+        logger.error('serviceIndex is not int')
+        return f'serviceIndex is not int', 400
+    if not isinstance(data['trustedAlgorithms'], list):
+        logger.error('trustedAlgorithms is not list')
+        return f'trustedAlgorithms is not list', 400
+    try:
+        logger.info('Lets get did %s' % did)
+        _record = dao.get(did)
+        if _record is None:
+            return f'Cannot find did: {did} ', 404
+        if not can_update_did(_record, data['updated'], data['signature'], logger):
+            logger.error('Not allowed to update did')
+            return f'Not allowed to update this DID', 401
+        _record['updated'] = get_timestamp()
+        index = 0
+        for service in _record['service']:
+            if index == data['serviceIndex']:
```

Comment:
serviceIndex does not have to match the list index. We should compare service.index with the data['serviceIndex']